spock-core/src/main/java/org/spockframework/mock/runtime/ByteBuddyMockInteractionValidation.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.spockframework.mock.runtime;
+
+import org.junit.platform.commons.support.ModifierSupport;
+import org.spockframework.mock.IInvocationConstraint;
+import org.spockframework.mock.IMockInteraction;
+import org.spockframework.mock.IMockObject;
+import org.spockframework.mock.MockImplementation;
+import org.spockframework.mock.constraint.EqualMethodNameConstraint;
+import org.spockframework.mock.constraint.EqualPropertyNameConstraint;
+import org.spockframework.runtime.InvalidSpecException;
+import org.spockframework.util.ThreadSafe;
+
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.requireNonNull;
+import static org.spockframework.runtime.GroovyRuntimeUtil.GET;
+import static org.spockframework.runtime.GroovyRuntimeUtil.IS;
+import static org.spockframework.runtime.GroovyRuntimeUtil.propertyToMethodName;
+
+/**
+ * {@link ByteBuddyMockInteractionValidation} validates the {@link IMockInteraction} for {@link ByteBuddyMockMaker} mocks.
+ *
+ * <ul>
+ *   <li>Checks for method interactions on final methods</li>
+ *   <li>Checks for property interactions on final methods</li>
+ * </ul>
+ *
+ * <p>Implementation note: The {@link ByteBuddyMockFactory} create a single instance for each mocked class
+ * and the same validation is used for multiple mocks of the same class.</p>
+ *
+ * @author Andreas Turban
+ */
+@ThreadSafe
+final class ByteBuddyMockInteractionValidation implements IMockInteractionValidation {
```

Comment:
It would also work for the `cglib` mock maker, but I did not want to implement the static field cache in cglib, because it is already deprecated.

If you want that supported, let me know, than I will have a look.